test/integration-serial/serial_deploy_test.rb

```diff
@@ -525,12 +525,35 @@ def test_batch_dry_run_apply_failure_falls_back_to_individual_resource_dry_run_v
   end
 
   def test_batch_dry_run_apply_success_precludes_individual_resource_dry_run_validation
-    Krane::KubernetesResource.any_instance.expects(:validate_definition).with do |kwargs|
-      kwargs[:kubectl].nil? && !kwargs[:dry_run]
-    end
+    Krane::KubernetesResource.any_instance.expects(:validate_definition).times(0)
     deploy_fixtures("hello-cloud", subset: %w(secret.yml))
   end
 
+  def test_resources_with_side_effect_inducing_webhooks_are_not_batched_server_side_dry_run
+    resp = JSON.parse(File.read(File.join(fixture_path("for_serial_deploy_tests"), "ingress_hook.json")))["items"]
+    Krane::ClusterResourceDiscovery.any_instance.expects(:fetch_mutating_webhook_configurations).returns(resp)
+    Krane::ResourceDeployer.any_instance.expects(:dry_run).with do |params|
+      params.length == 2 && (params.map(&:type) - ["Deployment", "Service"]).empty?
+    end
+    Krane::Ingress.any_instance.expects(:validate_definition)
+    result = deploy_fixtures('hello-cloud', subset: %w(web.yml.erb), render_erb: true)
+  end
+
+  def test_resources_with_side_effect_inducing_webhooks_with_transitive_dependency_fail_batch_running
+    resp = JSON.parse(File.read(File.join(fixture_path("for_serial_deploy_tests"), "secret_hook.json")))["items"]
+    Krane::ClusterResourceDiscovery.any_instance.expects(:fetch_mutating_webhook_configurations).returns(resp)
+    Krane::KubernetesResource.any_instance.expects(:validate_definition).times(4) # resources we expect to have to individually validate
+    deploy_fixtures('hello-cloud', subset: %w(web.yml.erb secret.yml), render_erb: true) do |fixtures|
+      container = fixtures['web.yml.erb']['Deployment'][0]['spec']['template']['spec']['containers'][0]
+      container['volumes'] = [
+        name: 'secret',
+        secret: {
+          secretName: fixtures['secret.yml']["Secret"][0]['metadata']['name'],
+        },
+      ]
```

Comment:
This _does not_ fail the dry-run, as I would have thought, even though the Secret resource is not part of the dry-run apply (and is not present on the cluster beforehand). I will try a few more failure scenarios to see if I can actually induce this type of transitive failure